extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +152,208 @@ export class Transport {
+            responseProcessor(
+                this.type_,
+                /** @type
+                 * {!../../../src/3p-analytics-common.AmpAnalytics3pResponse}
+                 */ (response));
+          }
+        },
+        true);
+    return frameData;
+  }
+
+  /**
+   * Called when a creative no longer needs its cross-domain iframe (for
+   * instance, because the creative has been removed from the DOM).
+   * Once all creatives using a frame are done with it, the frame can be
+   * destroyed.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   */
+  static doneUsingCrossDomainIframe(ampDoc, transportOptions) {
+    const frameUrl = transportOptions['iframe'];
+    const frameData = Transport.getFrameData(frameUrl);
+    if (--(frameData.usageCount)) {
+      // Some other instance is still using it
+      return;
+    }
+    ampDoc.body.removeChild(frameData.frame);
+    frameData.responseMessageUnlisten();
+    delete Transport.crossDomainIframes_[frameUrl];
+  }
+
+  /**
+   * Returns whether a url of a cross-domain frame is already known
+   * @param {!string} frameUrl
+   * @return {!boolean}
+   * @VisibleForTesting
+   */
+  static hasCrossDomainIframe(frameUrl) {
+    return hasOwn(Transport.crossDomainIframes_, frameUrl);
+  }
+
+  /**
+   * Create a unique value to differentiate messages from
+   * this particular creative to the cross-domain iframe
+   * @returns {string}
+   * @private
+   */
+  static createUniqueId_() {
+    return String(++(Transport.nextId_));
+  }
+
+  /**
+   * Sends an Amp Analytics trigger event to a vendor's cross-domain iframe,
+   * or queues the message if the frame is not yet ready to receive messages.
+   * @param {!string} event A string describing the trigger event
+   * @param {!string} frameUrl The URL of the iframe
+   * @VisibleForTesting
+   */
+  sendRequestUsingCrossDomainIframe(event, frameUrl) {
+    const frameData = Transport.getFrameData(frameUrl);
+    dev().assert(frameData, 'Trying to send message to non-existent frame');
+    dev().assert(frameData.queue,
+        'Event queue is missing for ' + this.id_);
+    frameData.queue.enqueue(this.id_, event);
+  }
+
+  /**
+   * Gets the FrameData associated with a particular cross-domain frame URL.
+   * @param frameUrl
+   * @returns {FrameData}
+   * @VisibleForTesting
+   */
+  static getFrameData(frameUrl) {
+    return Transport.crossDomainIframes_[frameUrl];
+  }
+
+  /**
+   * Removes all knowledge of cross-domain iframes.
+   * Does not actually remove them from the DOM.
+   * @VisibleForTesting
+   */
+  static resetCrossDomainIframes() {
+    Transport.crossDomainIframes_ = {};
+  }
+
+  /**
+   * @returns {!string} Unique ID of this instance of Transport
+   * @VisibleForTesting
+   */
+  getId() {
+    return this.id_;
+  }
 }
 
+/** @private {Object<string,FrameData>} */
```

Comment:
According to http://2ality.com/2015/02/es6-classes-final.html, class declarations are not hoisted. So I suspect that something like this won't work:
```
Foo.bar = 5;
class Foo {
  ...
}
```
But...giving it a try and will check in if it works.